### DISCUSSION TITLE
Trying to run argo workflow from a selected step

### DISCUSSION THREAD

**Discussion by Rachel-Rabinowitz** [argoproj/argo-workflows#9152] [Q&A] [answered]
I have this pipeline configuration:
`@dsl.pipeline(name='hello-workflow', description='Learning argo workflow')
def pipeline(message1: str = 'message1', message2: str = 'message2', message3: str = 'message3'):
    step1 = HelloWorflowStep1(message=message1)
    step2 = HelloWorflowStep2(message=message2)
    step3 = HelloWorflowStep3(message=message3).after(step2)`

I'm trying to run the following command:
`argo submit --from workflowtemplate/hello-workflow --entrypoint=hello-workflow-step2 --kubeconfig {KUBE_CONFIG_PATH} --namespace {namespace}`
`hello-workflow-step2` is the name of step2,
Only step2 running but I expect  step3 to run also

Any idea what that happens or any idea how to run a workflow from middle step

**Comment by terrytangyuan** [comment, marked as answer]
Duplicated https://github.com/argoproj/argo-workflows/issues/9160

**Comment by Rachel-Rabinowitz** [comment]
I opened it as a discussion and decided to open it as an issue
so it is duplicated
Can you answer me here
<https://github.com/argoproj/argo-workflows/issues/9160> ?
Thanks!
